projects/plugins/jetpack/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -750,15 +750,7 @@ public function update_data( $request ) {
 								$updated = new WP_Error( 'instant_search_disabled', 'Instant Search is disabled', array( 'status' => 400 ) );
 								$error   = $updated->get_error_message();
 							} else {
-								$instance = Automattic\Jetpack\Search\Instant_Search::instance();
-
-								// Perform initialization. This should have already been executed in modules/search.php.
-								// In other words: This is a failsafe.
-								if ( ! $instance ) {
-									Automattic\Jetpack\Search\Jetpack_Initializer::initialize();
-									$instance = Automattic\Jetpack\Search\Instant_Search::instance();
-								}
-
+								$instance = Automattic\Jetpack\Search\Instant_Search::instance( Jetpack_Options::get_option( 'id' ) );
```

Comment:
I guess the code might not be needed on WPCOM at all, but probably better be compatible as well to get the blog_id?